monkey/infection_monkey/master/agent_lifecycle.py

```diff
@@ -0,0 +1,10 @@
+from infection_monkey.island_api_client import IIslandAPIClient
+
+
+class AgentLifecycle:
+    def __init__(self, island_api_client: IIslandAPIClient):
+        self._island_api_client = island_api_client
+
+    def should_agent_stop(self) -> bool:
+        agent_signals = self._island_api_client.get_agent_signals()
+        return agent_signals.terminate is not None
```

Comment:
I don't think this component makes any sense. We've taken a thin component with very little functionality (IControlClient) and replaced it with an even thinner component with less functionality (AgentLifecycle). The two lines in this function can just be inlined into `AutomatedMaster._check_for_stop()`.

Alternatively, we could add `IIslandAPIClient.terminate_signal_set()` as a convenience function.